autobahn/twisted/websocket.py

```diff
@@ -75,6 +81,144 @@
 )
 
 
+def create_client_agent(reactor):
+    """
+    :returns: an instance implementing IWebSocketClientAgent
+    """
+    return _TwistedWebSocketClientAgent(reactor)
+
+
+def check_transport_config(transport_config):
+    """
+    raises a ValueError if `transport_config` is invalid
+    """
+    # XXX move me to "autobahn.websocket.util"
+    if not isinstance(transport_config, six.text_type):
```

Comment:
I guess this is partly about compatibility with existing autobahn APIs ... but sure would be nice to have a structured URI type instead of a string :/